postgresql-11.5/src/bin/psql/commandbteq.c

```diff
@@ -345,43 +347,29 @@ static dotResult
 exec_command_set(BteqScanState scan_state, bool active_branch)
 {
     bool        success = true;
-
+    char        quote;
     if (active_branch)
     {
-        char       *opt0 = bteq_scan_dot_option(scan_state,
-                                                  OT_BTEQ_NORMAL, NULL, false);
-
-        if (!opt0)
+        char *opt0 = read_connect_arg(scan_state);
+        if (opt0)
         {
```

Comment:
What will happen if the `opt0` is `NULL`? Please, cover that case as well according to the original BTEQ's behavior.

Reply:
the code blocks similar to the following:
```
if (width_keyword != NULL) {
// Lots of staff to do
} else {
printf("error message");
}
```
can be optimized as follows:
```
if (width_leyword == NULL) {
printf("error message");
return ...;
}

// lots of staff to do, as this line is equivalent to the else block
```

Please, update your code as required to increase readability and avoid nested if ... else ... clauses.